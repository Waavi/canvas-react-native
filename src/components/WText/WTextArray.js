import React from 'react'
import PropTypes from '#propTypes'
import { Text } from 'react-native'
import { WText } from './WText'

PropTypes.propTypes = {
	cls: PropTypes.string,
	texts: PropTypes.arrayOf(
		PropTypes.shape({
			bold: PropTypes.bool,
			italic: PropTypes.bool,
			green: PropTypes.bool,
			text: PropTypes.string.isRequired,
		})
	).isRequired,
}
PropTypes.defaultProps = {
	cls: undefined,
}
// eslint-disable-next-line
export function WTextArray({ cls, texts, children, ...restProps }) {
	return <Text {...restProps}>{texts.map((item, index) => textItem(cls, item, index))}</Text>
}

function textItem(cls, item, index) {
	let computedCls = cls || ''
	if (item.bold) {
		computedCls += ' semibold'
	}
	if (item.italic) {
		computedCls += ' italic'
	}
	if (item.green) {
		computedCls += ' green'
	}
	return (
		<WText key={index} cls={computedCls}>
			{item.text}
		</WText>
	)
}
