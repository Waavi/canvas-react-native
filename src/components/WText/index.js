export * from './WText'
export * from './WTextArray'
