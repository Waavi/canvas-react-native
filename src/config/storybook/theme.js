import { create } from '@storybook/theming'

export default create({
	// Is this a 'light' or 'dark' theme?
	base: 'light',

	// Color palette
	colorPrimary: 'red', // primary color
	colorSecondary: 'pink', // secondary color

	// UI
	appBg: 'papayawhip',
	appContentBg: 'white',
	appBorderColor: 'rgba(0,0,0,.1)',
	appBorderRadius: 4,

	// Fonts
	fontBase: '"Helvetica", Arial, sans-serif',
	fontCode: 'Monaco, monospace',

	// Text colors
	textColor: '#FFFFFF',
	textInverseColor: '#333333',

	// Toolbar default and active colors
	barTextColor: '#999999',
	barSelectedColor: 'blue',
	barBg: 'white',

	// Form colors
	inputBg: 'white',
	inputBorder: 'rgba(0,0,0,.1)',
	inputTextColor: '#333333',
	inputBorderRadius: 4,

	// Brand logo/text
	brand: `<svg .../>`,
})
