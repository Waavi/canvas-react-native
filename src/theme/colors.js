const basics = {
	lightgreen: '#18E2A5',
	darkgreen: '#23AE85',
	secondaryLightgreen: '#B1DCCD',
	secondaryDarkgreen: '#2E7D7F',
	lightorange: '#FBCC7C',
	darkorange: '#FF934B',
	lightred: '#F4573C',
	darkred: '#ED2434',
	lightyellow: '#FCFCD3',
	darkyellow: '#F1D833',

	white: '#FFF',
	lightestgray: '#F4F6FA',
	lightergray: '#EDF0F4',
	lightgray: '#C8CED8',
	mediumgray: '#A3ADBA',
	darkgray: '#5B6674',
	darkergray: '#3A434F',
	darkestgray: '#2b303c',
	black: '#222',
	totalblack: '#000',
}

const green = basics.lightgreen
const yellow = basics.darkyellow
const dark = basics.darkergray
const red = basics.darkred

export const Colors = {
	...basics,
	green,
	yellow,
	dark,
	red,

	primary: green,
	brand: green,

	sectionTitle: {
		background: '#08D79a',
		text: 'rgb(50, 120, 90)',
	},

	transparent: 'transparent',

	separator: '#dedde0', //'#f5f5f6',

	background: basics.lightergray,

	text: {
		totalBlack: basics.totalblack,
		black: basics.black,
		darkestgray: basics.darkestgray,
		darkgray: basics.darkgray,
		secondary: basics.mediumgray,
		label: basics.mediumgray,
		hint: basics.lightgray,
		'hint-dark': basics.darkgray,
		white: basics.white,
		brand: green,
		green,
		yellow,

		inverse: basics.white,
		inverseSecondary: basics.mediumgray,

		link: '#0088ff',
		error: red,
	},

	card: {
		subscribed: green,
		notSubscribed: basics.lightgray,
	},

	promo: {
		reward: green,
		points: green,
		pointsLight: '#70f0c9',
		offer: '#95D6C7',
		offerLight: '#dff2ee',
		stamp: '#FABA59',
		stampLight: '#fde3bc',
		stamps: '#FABA59',
		stampsLight: '#fde3bc',
		coupon: '#EECB19',
		couponLight: '#f5e078',
		couponp: '#EECB19',
		couponpLight: '#f5e078',
		bono: '#FA4C37',
		bonoLight: '#fca59b',
	},

	rating: {
		gray: '#dce1e6',
		darkgray: basics.darkgray,
		green,
		black: basics.black,
	},

	catalog: {
		background: '#3e4854',
		gray: '#647685',
		title: '#14e5a9',
		price: '#f9da31',
		thumbup: '#14e5a9',
		thumbdown: '#fe6448',
	},

	txnResult: {
		promo: basics.darkorange,
		rule: green,
	},
}
