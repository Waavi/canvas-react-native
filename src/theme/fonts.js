const type = {
	/*
		https://medium.com/@danielskripnik/how-to-add-and-remove-custom-fonts-in-react-native-b2830084b0e4

		How to install / uninstall fonts (iOS & Android):
        1. Add (or remove) them into @/assets/fonts
        2. Be sure they're named exactly as their Full Name (I think it should be done
            by their PostScript Name)
        3. run "react-native link assets"
    */
	bold: 'Ubuntu-Bold',
	italic: 'Ubuntu-Italic',
	light: 'Ubuntu-Light',
	medium: 'Ubuntu-Medium',
	regular: 'Ubuntu-Regular',
	semibold: 'Ubuntu-Medium',
}

const size = {
	8: 8,
	9: 9,
	10: 10,
	11: 11,
	12: 12,
	14: 14,
	16: 16,
	18: 18,
}

export const Fonts = {
	type,
	size,
}
