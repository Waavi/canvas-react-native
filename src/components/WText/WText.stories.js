import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { boolean } from '@storybook/addon-knobs'
import { StoryView, StoryHeader, StoryCase } from '@modules/react-native-storybook/components'
import { View, StyleSheet } from 'react-native'
import { Fonts, Colors } from '@/theme'
import { SPACINGS_ABSOLUTE, SPACINGS_LIST } from '@/theme/cls/spacingClsStyles'
import { WText } from './WText'

storiesOf('WText', module)
	.add('font sizes', () => {
		const fontSizes = [undefined, 10, 12, 14, 16, 18, 20, 24, 28, 35]
		const scaled = boolean('Scaled', true)
		return (
			<StoryView align="stretch" padded>
				{fontSizes.map(size => (
					<View style={styles.wrapper} key={`example-${size}`}>
						<WText size={size} scaled={scaled}>{`WText with size ${size}`}</WText>
					</View>
				))}
			</StoryView>
		)
	})
	.add('font types (families)', () => {
		const fontTypes = Object.keys(Fonts.type)
		return (
			<StoryView align="stretch" padded>
				{fontTypes.map(type => (
					<View key={type} style={styles.wrapper}>
						<WText size={20} cls={type}>{`WText with cls '${type}'`}</WText>
					</View>
				))}
			</StoryView>
		)
	})
	.add('font colors', () => {
		const colors = Object.keys(Colors.text).map(color => ({
			color,
			style: {
				backgroundColor:
					color === 'white' || color === 'inverse' ? Colors.black : Colors.transparent,
			},
		}))
		return (
			<StoryView align="stretch" padded>
				{colors.map(({ color, style }) => (
					<View key={color} style={[styles.wrapper, style]}>
						<WText size={20} cls={color}>{`WText with cls '${color}'`}</WText>
					</View>
				))}
			</StoryView>
		)
	})
	.add('alignment', () => {
		const alignments = ['left', 'center', 'right']
		return (
			<StoryView align="stretch" padded>
				{alignments.map(align => (
					<View key={align} style={styles.wrapper}>
						<WText size={20} cls={align}>{`WText with cls '${align}'`}</WText>
					</View>
				))}
			</StoryView>
		)
	})
	.add('flex', () => {
		const items = ['one', 'two', 'fourteen']
		const flexWrapper = {
			flexDirection: 'row',
			justifyContent: 'space-between',
		}
		const textStyle = {
			backgroundColor: 'yellow',
			borderColor: 'black',
			borderWidth: 1,
			textAlign: 'center',
		}
		const example = cls => (
			<StoryCase text={cls ? `Each item with cls '${cls}'` : 'default'}>
				<View style={flexWrapper}>
					{items.map(item => (
						<WText key={item} cls={cls} size={20} style={textStyle}>
							{item}
						</WText>
					))}
				</View>
			</StoryCase>
		)
		return (
			<StoryView align="stretch">
				{example(undefined)}
				{example('flex-0')}
				{example('flex-1')}
				{example('grow-0')}
				{example('grow-1')}
			</StoryView>
		)
	})
	.add('margins & paddings', () => {
		const textStyle = {
			backgroundColor: 'yellow',
			borderColor: 'black',
			borderWidth: 1,
		}
		const example = cls => (
			<StoryCase text={cls ? `with cls '${cls}'` : 'default'}>
				<WText cls={cls} size={16} style={textStyle}>
					Lorem ipsum
				</WText>
			</StoryCase>
		)
		return (
			<StoryView scrollable>
				<StoryHeader.h1 text="Margins" />
				<StoryHeader.h2
					text="Choose the margin"
					description="You can choose to add a margin to all sides or select one of them"
				/>
				{example(undefined)}
				{example('m-10')}
				{example('ml-10')}
				{example('mr-10')}
				{example('mt-10')}
				{example('mb-10')}

				<StoryHeader.h2
					text="Absolute margin values"
					description={`You have the following values: ${SPACINGS_ABSOLUTE.join(', ')}`}
				/>
				{example('m-5')}
				{example('m-10')}
				{example('m-20')}
				{example('m-30')}

				<StoryHeader.h2
					text="Margin values based on default spacings"
					description={`You have the following values: ${SPACINGS_LIST.join(', ')}`}
				/>
				{example('m-s')}
				{example('m-2s')}
				{example('m-3s')}

				<StoryHeader.h1 text="Paddings" />
				<StoryHeader.h2
					text="Choose the padding"
					description="You can choose to add a padding to all sides or select one of them"
				/>
				{example(undefined)}
				{example('p-10')}
				{example('pl-10')}
				{example('pr-10')}
				{example('pt-10')}
				{example('pb-10')}

				<StoryHeader.h2
					text="Absolute padding values"
					description={`You have the following values: ${SPACINGS_ABSOLUTE.join(', ')}`}
				/>
				{example('p-5')}
				{example('p-10')}
				{example('p-20')}
				{example('p-30')}

				<StoryHeader.h2
					text="Padding values based on default spacings"
					description={`You have the following values: ${SPACINGS_LIST.join(', ')}`}
				/>
				{example('p-s')}
				{example('p-2s')}
				{example('p-3s')}
			</StoryView>
		)
	})

const styles = StyleSheet.create({
	wrapper: {
		marginVertical: 5,
	},
})
