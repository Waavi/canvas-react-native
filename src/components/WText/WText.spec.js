import React from 'react'
import { mount, shallow } from 'enzyme'
import { Text, View } from 'react-native'
import { Fonts } from '@/theme'
import { WText } from './WText'

describe('Components/WText', () => {
	it('should be a wrapper of native Text', () => {
		const props = { children: 'text', numberOfLines: 1, onPress: () => {} }
		const wrapper = shallow(<WText {...props} />)
		const innerText = wrapper.find(Text)
		expect(innerText.props()).toMatchObject(props)
	})
	it('should set a base style', () => {
		const wrapper = shallow(<WText />)
		const innerStyle = wrapper.find(Text).props().style
		expect(innerStyle).toBeTruthy()
		expect(innerStyle).toMatchSnapshot()
	})
	it('should have a "class" for each font family defined at Fonts catalog', () => {
		const wrapper = mount(
			<View>
				{Object.keys(Fonts.type).map(type => (
					<WText key={type} cls={Fonts.type[type]}>
						test
					</WText>
				))}
			</View>
		)
		expect(wrapper).toMatchSnapshot()
	})
	it('should have a "class" for each font size defined at Fonts catalog', () => {
		const wrapper = mount(
			<View>
				{Object.keys(Fonts.size).map(size => (
					<WText key={size} cls={Fonts.size[size]}>
						test
					</WText>
				))}
			</View>
		)
		expect(wrapper).toMatchSnapshot()
	})
	it('should have the "classes": "title", "text", ...', () => {
		const classes = ['title', 'text', 'grow', 'left', 'center', 'right']
		const wrapper = mount(
			<View>
				{classes.map(cls => (
					<WText key={cls} cls={cls}>
						test
					</WText>
				))}
			</View>
		)
		expect(wrapper).toMatchSnapshot()
	})
})
