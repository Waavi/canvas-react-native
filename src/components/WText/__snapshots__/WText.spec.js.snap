// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/WText should have a "class" for each font family defined at Fonts catalog 1`] = `
<View>
  <View>
    <_class
      cls="Montserrat-Thin"
      key="thin"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-ExtraLight"
      key="extralight"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-Light"
      key="light"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-Regular"
      key="regular"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-Medium"
      key="medium"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-SemiBold"
      key="semibold"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-Bold"
      key="bold"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="Montserrat-Regular"
      key="italic"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="VarelaRound-Regular"
      key="varela"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
  </View>
</View>
`;

exports[`Components/WText should have a "class" for each font size defined at Fonts catalog 1`] = `
<View>
  <View>
    <_class
      cls={8}
      key="8"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={9}
      key="9"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={10}
      key="10"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={11}
      key="11"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={12}
      key="12"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={14}
      key="14"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={16}
      key="16"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls={18}
      key="18"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
  </View>
</View>
`;

exports[`Components/WText should have the "classes": "title", "text", ... 1`] = `
<View>
  <View>
    <_class
      cls="title"
      key="title"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="text"
      key="text"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="grow"
      key="grow"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
              "color": "#222",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16,
              "lineHeight": 20,
            },
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="left"
      key="left"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
            ],
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#222",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
              ],
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="center"
      key="center"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "center",
              },
            ],
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#222",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "center",
                },
              ],
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
    <_class
      cls="right"
      key="right"
      scaled={false}
      size={16}
    >
      <Text
        scaled={false}
        size={16}
        style={
          Array [
            undefined,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#222",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "right",
              },
            ],
            Object {
              "fontSize": 16,
              "lineHeight": 20,
            },
            undefined,
          ]
        }
      >
        <Text
          scaled={false}
          size={16}
          style={
            Array [
              undefined,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#222",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "right",
                },
              ],
              Object {
                "fontSize": 16,
                "lineHeight": 20,
              },
              undefined,
            ]
          }
        >
          test
        </Text>
      </Text>
    </_class>
  </View>
</View>
`;

exports[`Components/WText should set a base style 1`] = `
Array [
  undefined,
  Object {
    "backgroundColor": "transparent",
    "color": "#222",
    "fontFamily": "Montserrat-Regular",
    "fontSize": 16,
    "lineHeight": 20,
  },
  Object {
    "fontSize": 16,
    "lineHeight": 20,
  },
  undefined,
]
`;
